lambda_functions/nightly_stats_bigquery_upload/nightly_stats_bigquery_upload_lambda.py

```diff
@@ -76,26 +108,49 @@ def add_updated_rows_for_date(
 def lambda_handler(
     event,
     _context,
-):
+) -> dict[str, int]:
+    logger.debug('get credentials . . .')
     service_account_info = read_service_account_info_from_ssm()
     credentials = service_account.Credentials.from_service_account_info(service_account_info)
+    logger.debug('. . . credentials obtained')
 
+    logger.debug('creating bigquery client . . .')
     bigquery_client = bigquery.Client(credentials=credentials)
+    logger.debug('. . . bigquery client created')
 
-    table_id = f'vsp-analytics-and-insights.platform_vanotify.{os.getenv("ENVIRONMENT")}-statistics'
-
+    logger.debug('getting bucket name and object key . . .')
     bucket_name = get_bucket_name(event)
     object_key = get_object_key(event)
+    logger.debug('. . . bucket name and object key obtained')
+
+    # get the date and data_type (billing or stats) from the s3 object key
+    date, data_type, _ext = object_key.split('.')
 
+    # determine the table id based on the data_type
+    if data_type == 'stats':
+        table_id = TABLE_ID_STATS
+    elif data_type == 'billing':
+        table_id = TABLE_ID_BILLING
+    else:
+        raise ValueError(f'Unexpected data type, expected "stats" or "billing" got: "{data_type}"')
+
+    logger.debug('checking if table exists . . .')
     try:
         bigquery_client.get_table(table_id)
     except NotFound:
-        pass
-    else:
-        delete_existing_rows_for_date(bigquery_client, table_id, object_key)
+        raise
```

Comment:
Remove the "except" block if all it should do is raise.

Reply:
I meant to put a log statement before that to clearly state what's happening so we don't have to look through a stack trace to figure it out. Good catch